lib/web3/httpprovider.js

```diff
@@ -36,13 +36,16 @@ if (typeof window !== 'undefined' && window.XMLHttpRequest) {
 }
 
 var XHR2 = require('xhr2'); // jshint ignore: line
+var btoa = require('btoa');
```

Comment:
Http Provider will be deprecated as websockets will be the standard.
Im not sure if a extra package is necessary when the user can simply add the user@pw to the URL.